thirdparty/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 add_subdirectory(cJSON)
 if (NOT SMTK_USE_SYSTEM_MOAB)
+  set(save_CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+  set(save_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  set(save_CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
   add_subdirectory(moab)
+  set(CMAKE_C_FLAGS "${save_CMAKE_C_FLAGS}" CACHE STRING "" FORCE)
+  set(CMAKE_CXX_FLAGS "${save_CMAKE_CXX_FLAGS}" CACHE STRING "" FORCE)
+  set(CMAKE_Fortran_FLAGS "${save_CMAKE_Fortran_FLAGS}" CACHE STRING "" FORCE)
```

Comment:
Any chance of fixing MOAB instead? As I read this, it prevents users from specifying flags on SMTK because MOAB misbehaves.